lib/samples/filter_features_in_scene/filter_features_in_scene.dart

```diff
@@ -0,0 +1,259 @@
+  // adds a red outline graphic, and sets up a filter for OSM buildings.
+  void _createAndFilterPolygon() {
+    // Build a polygon from the full extent of the San Francisco buildings layer.
+    final polygonBuilder = PolygonBuilder(
+      spatialReference: _sceneViewController.spatialReference,
+    );
+    final extent = _sfBuildingsSceneLayer.fullExtent!;
+    polygonBuilder.addPointXY(x: extent.xMin, y: extent.yMin);
+    polygonBuilder.addPointXY(x: extent.xMax, y: extent.yMin);
+    polygonBuilder.addPointXY(x: extent.xMax, y: extent.yMax);
+    polygonBuilder.addPointXY(x: extent.xMin, y: extent.yMax);
+
+    // Convert the polygon builder to a geometry.
+    _sceneLayerExtentPolygon = polygonBuilder.toGeometry();
+
+    // Create a red outline symbol with transparent fill.
+    final outlineSymbol = SimpleFillSymbol(
+      color: Colors.transparent,
+      outline: SimpleLineSymbol(color: Colors.red, width: 5),
+    );
+
+    // Create a graphic using the polygon geometry and symbol.
+    _sfGraphic = Graphic(
+      geometry: polygonBuilder.toGeometry(),
+      symbol: outlineSymbol,
+    );
+
+    // Create the SceneLayerPolygonFilter to later apply to the OSM buildings layer.
+    _sceneLayerPolygonFilter = SceneLayerPolygonFilter(
+      polygons: [polygonBuilder.toGeometry() as Polygon],
+      spatialRelationship: SceneLayerPolygonFilterSpatialRelationship.disjoint,
+    );
+  }
+
+  // Filter scene based on the current filter action.
+  void onSceneActionPressed() {
+    switch (_sceneFilterAction) {
+      case SceneFilterAction.load:
+        _addBuildings();
+      case SceneFilterAction.filter:
+        _filterScene();
+      case SceneFilterAction.reset:
+        _resetScene();
+    }
+    setState(() => _sceneFilterAction = _sceneFilterAction.next());
+  }
+
+  // Add the San Francisco buildings scene layer and its extent graphic to the scene.
+  void _addBuildings() {
+    _sceneViewController.arcGISScene?.operationalLayers.add(
+      _sfBuildingsSceneLayer,
+    );
+    _sfGraphicsOverlay.graphics.add(_sfGraphic);
+  }
+
+  // Apply a polygon filter to hide OSM buildings within the San Francisco extent.
+  void _filterScene() {
+    // If no filter is set, assign the polygon filter.
+    if (_osmBuildingsSceneLayer.polygonFilter == null) {
+      _osmBuildingsSceneLayer.polygonFilter = _sceneLayerPolygonFilter;
+    }
+    // If the filter exists but has no polygons, add the extent polygon.
+    else {
+      _sceneLayerPolygonFilter.polygons.add(
+        _sceneLayerExtentPolygon as Polygon,
+      );
+    }
+  }
+
+  // Reset the scene by removing layers, filters, and graphics.
+  void _resetScene() {
+    _sceneViewController.arcGISScene!.operationalLayers.clear();
+    _osmBuildingsSceneLayer.polygonFilter?.polygons.clear();
+    _sfGraphicsOverlay.graphics.clear();
+  }
+}
+
+// The different states for filtering features in a scene.
+enum SceneFilterAction {
+  load('Load San Francisco buildings'),
```

Comment:
```suggestion
  load('Load detailed buildings'),
```
To match instructions in readme